api/server/router/network/network_routes.go

```diff
@@ -79,8 +80,16 @@ func (n *networkRouter) postNetworkCreate(ctx context.Context, w http.ResponseWr
 		return err
 	}
 
-	if _, err := n.clusterProvider.GetNetwork(create.Name); err == nil {
-		return libnetwork.NetworkNameError(create.Name)
+	if nw, err := n.clusterProvider.GetNetwork(create.Name); err == nil {
```

Comment:
Makes sense, however, we need to check the conflict before calling `backend.CreateNetwork()`, so as to detect a conflict like this:
```console
$  docker service create -d overlay foo
$  docker service create -d bridge foo 
Error response from daemon: network with name foo already exists
```

A solution I can come up with is to call `clusterProvider.CreateNetwork()` before calling `backend.CreateNetwork()` for swarm-scoped networks, but I'm not sure how we can detect the scope here.

WDYT?
cc @mrjana @mavenugo @aboch